Fix `initial` and `auto` value for flex

iegik opened the pull request:
This PR to handle the `initial` and `auto` value for `flex` CSS property

`flex: initial → flex: 0 1 auto`
`flex: auto → flex: 1 1 auto`

https://github.com/luisrudge/postcss-flexbugs-fixes/issues/48

iegik commented:
tests: https://github.com/luisrudge/postcss-flexbugs-fixes/pull/55

luisrudge commented:
Hi, thanks for the PR. Can you add a bit of context on this change? 😬 

[This](https://developer.mozilla.org/en-US/docs/Web/CSS/flex#Values) says the default is 0

iegik commented:
Ok, I leave default value to 'auto', but also leave 0 if it appears.

luisrudge commented:
I guess this conflicts with https://github.com/luisrudge/postcss-flexbugs-fixes/pull/22

@silvenon what do you think about this PR? 🤔

silvenon commented:
Initial value is not the same as default value. The default value for `flex-basis` is the one being set when you're using the `flex` shorthand without covering `flex-basis`, initial value is what was there initially.

  - `flex: initial` → `flex: 0 1 auto`
  - `flex: 1` → `flex: 1 1 0%`

Maybe you could add a transformation for `flex: initial` and modify the readme to recommend globally setting `flex: initial` instead of `flex-basis: 1`. I don't know if this plugin already has that transformation, it might be handy.

iegik commented:
@silvenon what about "Safari seems to hate '0%' and the others seems to make do with a nice value when basis is missing" ?
IE11 waits 3 props, not 2. ('flex: 1 → flex: 1 1')

Should we normalize `flexBasis` to `0`?

luisrudge commented:
@iegik are you talking about this issue? https://github.com/luisrudge/postcss-flexbugs-fixes/issues/48

If so, let's fix both scenarios but in different PRs. Please update this PR to handle only the 'initial' value (and add a test for it), and let's create another PR that fixes #48 

Thanks 🎉

luisrudge commented:
@iegik thanks! can you add a test for this, please? 🎉

iegik commented:
thanks

iegik commented:
breaks 8.1.a

iegik commented on index.js:
```diff
@@ -3,7 +3,7 @@ var bug4 = require('./bugs/bug4');
 var bug6 = require('./bugs/bug6');
 var bug81a = require('./bugs/bug81a');
 
-var doNothingValues = ['none', 'auto', 'content', 'inherit', 'initial', 'unset'];
```
Bad practice to declarate rule in index.
These checks chould be repeated in each ./bugs/bunny file!

axelboc commented:
@iegik I see you've renamed the PR to focus on [flexbug #8](https://github.com/philipwalton/flexbugs/blob/master/README.md#flexbug-8), but the code still seems to do a lot more. May I ask where you're at? It's now a fairly simple problem, which shouldn't affect the other bugs. Let me know if you need a hand ;)

iegik commented:
@axelboc No, this PR is focused to [FlexBug #6](https://github.com/philipwalton/flexbugs/blob/master/README.md#flexbug-6)

- added default values for `initial` and `auto`
- removed tests, that not affects to this bug

The problem now is, that changing default `flex` values in **bug 6** also affects to **bug 4** and **8.1.a** tests.

iegik commented:
I think, that this PR and https://github.com/luisrudge/postcss-flexbugs-fixes/pull/56 should be merged as one PR
